src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -81,6 +101,31 @@ protected override void Shutdown()
 
         #region Built-in Message Handlers
 
+        protected async Task HandleConfigurationDoneRequest(
+            object args,
+            RequestContext<object> requestContext)
+        {
+            // Ensure that only the second message between launch and
+            // configurationDone requests, actually launches the script.
+            lock (syncLock)
```

Comment:
Theoretically you shouldn't need a lock for this since all requests are handled serially by the MessageReader (a SynchronizationContext handles reads in a queue).  Did you run into a case where there was a race condition or are you just being cautious?